src/templates/BlogPost/BlogPost.js

```diff
@@ -16,34 +16,54 @@ import styles from './BlogPost.module.scss';
 const BlogPost = ({ pageContext }) => {
   const {
     data,
+    recommendedPosts,
     recent: { previous, next },
   } = pageContext;
+
+  //selection of recommended posts in the current post
+  const RECOMMENDED_POSTS_LIMIT = 3;
+  let filteredPosts = recommendedPosts.filter(post => post.id !== data.id); //trying get 3 post, where 'current post != recommended'
+  filteredPosts.length = RECOMMENDED_POSTS_LIMIT;
+  const [headerIsWhite, setHeaderIsWhite] = React.useState(false);
   const thumbnailsItems = [];
   if (next) thumbnailsItems.push(next);
   if (previous) thumbnailsItems.push(previous);
-
   const pageWrapperClass = classNames(styles.container, 'pageWrapper');
+  const articleRef = React.useRef(null);
+  function scrollHandler() {
+    const pos = articleRef.current.getBoundingClientRect();
+    if (pos.y <= 0 && -pos.y < pos.height) {
+      setHeaderIsWhite(true);
+      return;
+    }
 
+    setHeaderIsWhite(false);
+  }
+  React.useEffect(() => {
+    window.addEventListener('scroll', scrollHandler);
+    return () => window.removeEventListener('scroll', scrollHandler);
+  }, []);
   return (
     <Providers>
       <BackgroundStars />
-      <Layout>
+      <Layout headerIsWhite={headerIsWhite}>
         <Head>
-          <title>{data.title} - Halo Lab Blog</title>
-        </Head>
+          <title> {data.title} - Halo Lab Blog </title>{' '}
```

Comment:
why do we need spaces in this file? `{' '}`

Reply:
fixed - https://github.com/Halo-Lab/halo-lab/pull/104